packages/client/hmi-client/src/composables/useCharts.ts

```diff
@@ -652,10 +654,89 @@ export function useCharts(
 		return weightsCharts;
 	};
 
-	const useSimulateSensitivityCharts = (chartSettings: ComputedRef<ChartSetting[]>) => {
+	function createSensitivityBins(
+		records: Record<string, any>[],
+		selectedVariable: string,
+		numBins: number = 7
+	): Map<string, number[]> {
+		if (numBins < 1) {
+			throw new Error('Number of bins must be at least 1.');
+		}
+
+		// Extract the selected variable values and ensure they are numeric
+		const samples = records.map((record) => {
+			const value = record[selectedVariable];
+			const id = record.sample_id;
+			if (typeof value !== 'number') {
+				throw new Error(`The selected variable "${selectedVariable}" must map to numeric values.`);
+			}
+			if (typeof id !== 'number') {
+				throw new Error('The "sample_id" field must be a numeric value.');
+			}
```

Comment:
Is checking for sample_id to be numeric an actual thing we care about? eg when would it go wrong realistically? and can we save some computation and just skip the check?

Reply:
Nah there probably wouldn't be an issue, Ill remove it